app/src/main/java/com/rob729/quiethours/Adapter/ProfileListAdapter.kt

```diff
@@ -65,12 +69,50 @@ class ProfileListAdapter(
             binding.TxtImg.setText(item.name[0].toString())
             binding.TxtImg.avatarBackgroundColor = bgColors[item.colorIndex]
             binding.timeStamp.text = item.timeInstance
+            binding.pauseSwitch.isChecked = item.pauseSwitch
 
             binding.profileCard.setOnClickListener {
                 val args = Bundle()
                 args.putParcelable("Profile", item)
                 val dialog = DetailsFragment.newInstance(args)
-                dialog.show((parentView.context as FragmentActivity).supportFragmentManager, "DialogFragment")
+                dialog.show(
+                    (parentView.context as FragmentActivity).supportFragmentManager,
+                    "DialogFragment"
+                )
+            }
+            if (!item.pauseSwitch && item.ehr == Calendar.getInstance().get(Calendar.HOUR_OF_DAY) && item.emin == Calendar.getInstance().get(Calendar.MINUTE)) {
+                StoreSession.writeInt(AppConstants.BEGIN_STATUS, 0)
+                StoreSession.writeLong(AppConstants.ACTIVE_PROFILE_ID, 0)
+            }
```

Comment:
@rob729 I have used these lines to cover the condition mentioned by you Now the app is working fine...Please have a look

Reply:
@coder2699 Still not working as expected. Please test the app before requesting the review.

Reply:
@rob729 
I have tested the app for following cases :
1) Profile start time: 11:20 PM and end time: 11:22 PM. I pause the profile at 11:21 PM and then activate the profile at 11:24 PM. (Works as expected)
2) Profile start time: 11:20 PM and end time: 11:22 PM. I pause the profile at 11:15 PM and then activate the profile at 11:18 PM. Then profile gets activated at original start time i.e. 11:20PM.
3) Profile start time: 11:20 PM and end time: 11:25 PM. I pause the profile at 11:15 PM and then activate the profile at 11:22 PM. Then profile gets activated at 11:23PM ( as expected) .
4) Profile start time: 11:20 PM and end time: 11:25 PM. I pause the profile at 11:21 PM and then activate the profile at 11:23 PM. (Works as expected) 
5) Profile start time: 11:20 PM and end time: 11:22 PM. I did not pause/resume the profile and it works as expected. 

If I am missing out on any condition please let me know.

Reply:
<img src="https://media.giphy.com/media/r7gbr6tWBkMToDxi9P/giphy.gif" width="300"/>
@coder2699  Have a look at the above GIF. See the clock time and profile end time and check why this is happening?

Reply:
@rob729 I am unable to reproduce the same. Sorry to bother you for this, but it will be really helpful if you could tell what did u do so that u faced this?

Reply:
Make a profile, pause after it starts, and wait till it ends. After it ends then try resuming the profile.

Reply:
@rob729 thanks for your comment, I was able to get the problem you were facing. 

Actually If-else block is used to check the value stored in shared prefs and then the action to be performed is followed. For the values to be reflected one should restart the Main Fragment. 
You will not see this problem if after the end time you either go to any other fragment and come back to Main Fragment or if you close the app and reopen it.

Reply:
This is very similar to what happens when you create a profile. Say, if I create a profile for 8:25PM and when the New Profile fragment takes me to the MainFragment I remain there till the start time. When the time starts, the work starts but I won't see the currently active profile card (unless I go somewhere else and come back to MainFragment)

It happens because If-else block in this case, and in case of Pause/Resume gets executed only once when MainFragment starts.

Reply:
If you are trying to justify that this is not a bug, then please don't try justifying it.
This is a bug and it has to be solved. Moreover, the current profile case which you have told above is also a bug and not the expected behavior.
When a user will be using the app, they will not the things like they have to go to another screen or reopen the app for it to work fine. For them it will be a bug so these things have to be fixed.